contributed_definitions/nyaml/NXcoordinate_system_set.yaml

```diff
@@ -88,138 +88,259 @@ doc: |
   always guide the user for each group and field which of the coordinate system
   one refers to.
 type: group
+# express conventions explicitly and understandable
+# appdefs each instance of NXentry should have its own set of conventions
+# method-specific additional conventions on top of the ones formulated here
+# should go to method-specific base classes like NXcoordinate_system_em_ebsd
+# use depends_on field - not attribute - to point to conventions which in specific cases
 NXcoordinate_system_set(NXobject):
+  rotation_handedness(NX_CHAR):
```

Comment:
would that not be someting that is defined per coordinate system. Or do you want to say, we use the same convention for all CSs?

Reply:
I see your arguments and I agree they are substantiated yet I feel a bit like I am running in circles with this:

These five fields are specific concepts based on community suggested best practices how to describe 3D rotations as their docstrings specify, cleaner than most of our docstrings certainly in EM. Therefore, I see a substantiated argument to have them as an own group "say <<that community conventions>>" as this is semantically a category (conventions)

Now comes NeXus and their best practices, here I have two choices:
1a) Add this group of fields to an existent base class
1b) or make this group a own base class
2.) Have that defined on the fly i.e. in an appdef.

My original suggestion was 1b) the view of NeXus or at least our guess of NIAC point will be why to have this as an own base class, we do not wish that the number of base classes increases like crazy, I agree and at the same time I disagree because.
I would like to be able to follow a single source principle that means if I write an appdef
I dont want to repeat all these exact definitions again but just use them, i.e. use that <<conventions>> base class it is clearer it avoids spurious mistakes by copying over docstrings incorrectly and it keeps the appdef cleaner.

I agree that these sort of conventions are related only to the discussion of coordinate system.
Forget the symbols used for these fields for a sec and read only the docstring which is de facto concept. Then it is clear that it is not the handedness of an instance NXcoordinate_system but the handedness for a rotation about an axis an axis that can live in any 3D coordinate system.

An alternative would be to place these definitions to NXrotation_set where these would fill better. But used in practice then each NXrotation_set instance would need to have these definitions spelled out again or have pointer to it e.g. along the lines of a "depends_on" field etc or again some convention hidden in the NeXus docs. I dont want any of these later opinions. I think we can be sportive enough to substantiate why having even small well-defined base classes can be useful.

The true discussion in my opinion always oscillates between two extremes either put concept in base classes which ought not become too packed glossaries nor too domain specific or to put it in an appdef and then for any other appdef you have to start defining it again.

I think I should revert this to solution 1b and have NXcoordinate_system_rotation_convention.yaml

Reply:
As is said in my main statement above, I think keeping these conventions liker `rotation_handedness`, `rotation_convention`, `euler_angle_conventions`, and so on in NXcoordinate_system_set is a good idea. So I would advocate against `NXcoordinate_system_rotation_convention`´and just keep them in `NXcoordinate_system_set`. The main argument is that `NXcoordinate_system_set` is, among others, to be used to describe the community-agreed coordinate system, that it makes sense to bundle these with the community-agreed conventions.

So tldr, keep them here if you agree.

Reply:
These are also not domain-specific, so they deserve to be in this base class.